velox/dwio/parquet/reader/RepeatedColumnReader.cpp

```diff
@@ -108,21 +108,23 @@ void ensureRepDefs(
 }
 
 MapColumnReader::MapColumnReader(
-    std::shared_ptr<const dwio::common::TypeWithId> requestedType,
+    const std::shared_ptr<const dwio::common::TypeWithId>& requestedType,
+    const std::shared_ptr<const dwio::common::TypeWithId>& dataType,
     ParquetParams& params,
     common::ScanSpec& scanSpec)
     : dwio::common::SelectiveMapColumnReader(
```

Comment:
This is not in this PR, but could you please check SelectiveMapColumnReader's cstr?
```
SelectiveMapColumnReader::SelectiveMapColumnReader(
    const std::shared_ptr<const dwio::common::TypeWithId>& requestedType,
    const std::shared_ptr<const dwio::common::TypeWithId>& dataType,
    FormatParams& params,
    velox::common::ScanSpec& scanSpec)
    : SelectiveRepeatedColumnReader(
          dataType->type(),
          params,
          scanSpec,
          dataType),
      requestedType_{requestedType} {}
```
I see 2 issues: 
1) the types were not passed in correctly to SelectiveRepeatedColumnReader;
2) SelectiveMapColumnReader defines itself requestedType_ which may shadow the one in SelectiveColumnReader. Is this necessary?

Reply:
We can do this in a different PR.  Let's keep this one for parquet only.

Reply:
CC: @kevinwilfong